RecoLocalMuon/CSCRecHitD/src/CSCMake2DRecHit.cc

```diff
@@ -140,37 +140,44 @@ CSCRecHit2D CSCMake2DRecHit::hitFromStripAndWire(const CSCDetId& id,
   float sigmaWithinTheStrip = -99.;
   int quality = -1;
   LocalPoint lp0(0., 0.);
+  int wglo = wg_left;
+  int wghi = wg_right;
+  float ymiddle;
+
+  // First wire of first wiregroup in cluster
+  wglo = layergeom_->middleWireOfGroup(wglo) + 0.5 - 0.5 * layergeom_->numberOfWiresPerGroup(wglo);
+  // Last wire in last wiregroup of cluster (OK if only 1 wiregroup too)
+  wghi = layergeom_->middleWireOfGroup(wghi) - 0.5 + 0.5 * layergeom_->numberOfWiresPerGroup(wghi);
 
   float stripWidth = -99.f;
   // If at the edge, then used 1 strip cluster only
   if (centerStrip == 1 || centerStrip == specs_->nStrips() || nStrip < 2) {
-    lp0 = layergeom_->stripWireIntersection(centerStrip, centerWire);
+    lp0 = (layergeom_->possibleRecHitPosition(float(centerStrip) - 0.5, wglo, wghi)).first;
+    ymiddle = lp0.y();
     positionWithinTheStrip = 0.f;
     stripWidth = layergeom_->stripPitch(lp0);
     sigmaWithinTheStrip = stripWidth * inv_sqrt_12;
     quality = 2;
   } else {
     // If not at the edge, used cluster of size ClusterSize:
-    LocalPoint lp11 = layergeom_->stripWireIntersection(centerStrip, centerWire);
+    LocalPoint lp11 = (layergeom_->possibleRecHitPosition(float(centerStrip) - 0.5, wglo, wghi)).first;
+    ymiddle = lp11.y();
     stripWidth = layergeom_->stripPitch(lp11);
 
     //---- Calculate local position within the strip
     float xWithinChamber = lp11.x();
     quality = 0;
-    if (layergeom_->inside(lp11)) {  // save time; this hit is to be discarded anyway - see isHitInFiducial(...)
-
-      xMatchGatti_->findXOnStrip(id,
-                                 layer_,
-                                 sHit,
-                                 centerStrip,
-                                 xWithinChamber,
-                                 stripWidth,
-                                 tpeak,
-                                 positionWithinTheStrip,
-                                 sigmaWithinTheStrip,
-                                 quality);
-    }
-    lp0 = LocalPoint(xWithinChamber, layergeom_->yOfWire(centerWire, xWithinChamber));
+    xMatchGatti_->findXOnStrip(id,
```

Comment:
how frequently is this condition expected to fail?
The (removed) comment implies that the cost may be large.
If the cost is large, it would be better to find a way to not skip this if completely

Reply:
As far as I understand this we get rid of such cases when the wire+strip intersection gives us a local coordinate outside the layer's sensitive area (see slide 12 of presentation attached to this PR). That's why there is no need in this check. This is @ptcox suggestion for the code. If I am wrong, Tim, could you please comment?

Reply:
Sorry I only just saw I'd been asked to comment.
Yes, this 'inside'  test was a sanity check to catch edge cases where wire and strip intersection fell outside the chamber.  It was only an issue for ME1/1 extreme wire groups. The new code has this sanity check implicitly included (in CSCLayerGeometry::intersectionOfStripAndWire) so the test is unnecessary.